src/main/java/hillel/doctorRest/clinic/review/ReviewService.java

```diff
@@ -0,0 +1,83 @@
+package hillel.doctorRest.clinic.review;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import lombok.val;
+
+@Service
+@AllArgsConstructor
+
+public class ReviewService {
+
+    private final ReviewRepository reviewRepository;
+
+    public Review createReview(Review review) {
+        return reviewRepository.save(review);
+    }
+
+    public void saveReview(Review review) {
+        reviewRepository.save(review);
+    }
+
+    public Optional<Review> findById(Integer id) {
+        return reviewRepository.findById(id);
+    }
+
+    public List<Review> findAll() {
+        return reviewRepository.findAll();
+    }
+
+    public OptionalDouble averageService() {
+        return findAll().stream().map(review -> review.getService())
+                .flatMap(Optional::stream).mapToDouble(value -> value).average();
+    }
+
+    public OptionalDouble averageEquipment() {
+        return findAll().stream().map(review -> review.getEquipment())
+                .flatMap(Optional::stream).mapToDouble(value -> value).average();
+    }
+
+    public OptionalDouble averageQualificationSpecialist() {
+        return findAll().stream().map(review -> review.getQualificationSpecialist()
+        ).flatMap(Optional::stream).mapToDouble(value -> value).average();
+    }
+
+    public OptionalDouble averageEffectivenessOfTreatment() {
+        return findAll().stream().map(review -> review.getEffectivenessOfTreatment()
+        ).flatMap(Optional::stream).mapToDouble(value -> value).average();
```

Comment:
Usually it is formatted like:
```
return findAll().stream()
    .map(review -> review.getEffectivenessOfTreatment())
    .flatMap(Optional::stream)
    .mapToDouble(value -> value)
    .average();
```
Also in case if there are no votes, it is absolutely fine to return 0 instead of empty optional.